docs/calendar.md

```diff
@@ -0,0 +1,277 @@
+- a string that is accepted by `new Temporal.Calendar()`; or
+- a string in the ISO 8601 format.
+
+Note that the ISO 8601 string can be extended with a `[c=identifier]` annotation in square brackets appended to it.
+Without such an annotation, the calendar is taken to be `iso8601`.
+
+This function is often more convenient to use than `new Temporal.Calendar()` because it handles a wider range of input.
+
+Usage examples:
+```javascript
+// Calendar names
+cal = Temporal.Calendar.from('iso8601');
+cal = Temporal.Calendar.from('gregory');
+
+// ISO 8601 string with or without calendar annotation
+cal = Temporal.Calendar.from('2020-01-13T16:31:00.065858086');
+cal = Temporal.Calendar.from('2020-01-13T16:31:00.065858086-08:00[America/Vancouver][c=iso8601]');
+
+// Existing calendar object
+cal2 = Temporal.TimeZone.from(cal);
+
+/*⚠️*/ tz = Temporal.TimeZone.from('local');  // not a built-in calendar, throws
+/*⚠️*/ tz = Temporal.TimeZone.from({id: 'UTC'});  // not a Calendar object, throws
+/*⚠️*/ tz = Temporal.TimeZone.from('[c=iso8601]');  // lone annotation not a valid ISO 8601 string
+```
+
+## Properties
+
+### calendar.**id** : string
+
+The `id` property gives an unambiguous identifier for the calendar.
+Effectively, this is whatever `calendarIdentifier` was passed as a parameter to the constructor.
+
+## Methods
+
+### calendar.**year**(_date_: Temporal.Date) : number
+
+### calendar.**month**(_date_: Temporal.Date) : number
+
+### calendar.**day**(_date_: Temporal.Date) : number
+
+### calendar.**dayOfWeek**(_date_: Temporal.Date): number
+
+### calendar.**dayOfYear**(_date_: Temporal.Date): number
+
+### calendar.**weekOfYear**(_date_: Temporal.Date): number
+
+### calendar.**daysInMonth**(_date_: Temporal.Date): number
+
+### calendar.**daysInYear**(_date_: Temporal.Date): number
+
+### calendar.**isLeapYear**(_date_: Temporal.Date): boolean
+
+The above methods are all similar.
+They provide a way to query properties of a particular date in the calendar's date reckoning.
+
+**Parameters:**
+- `date` (`Temporal.Date`): A date
+
+**Returns:** some piece of data (year, month, day, etc., depending on the method) associated with `date`, in `calendar`'s calendar system.
+
+None of the above methods need to be called directly except in specialized code.
+They are called indirectly when reading the various properties of `Temporal.DateTime`, `Temporal.Date`, or `Temporal.YearMonth`.
+
+For example:
+```javascript
+const date = Temporal.Date.from('2020-05-29').withCalendar('hebrew');
+date.year  // => 5780
+date.calendar.year(date)  // same result, but calling the method directly
+date.daysInYear  // => 355
+date.calendar.daysInYear(date)  // same result, but calling the method directly
+```
+
+### calendar.**dateFromFields**(_fields_: object, _options_: object, _constructor_: function) : Temporal.Date
+
+### calendar.**dateTimeFromFields**(_fields_: object, _options_: object, _constructor_: function) : Temporal.DateTime
```

Comment:
We aren't supporting alternate ways of telling time.  Therefore, do we need the `dateTimeFromFields` method, or can we / should we use `dateFromFields`, and then concatenate the result with the time calculated using the normal `Time.from()` method?

Reply:
:bulb: I love it :bulb:

Reply:
It does, however, cause problems at the edge of the min/max range. (The calendar needs to be able to do RegulateDateTime on the whole thing.)

Reply:
I'm not sure there's a way around this since RegulateDate/RegulateDateTime etc. are fundamentally calendar-sensitive operations.

Reply:
End-of-range behavior is an edge case.  If we can find a workaround, we should not let edge cases deter API design.

You can always add or subtract 1 day from the ISO fields in order to get the correct adjustment with regard to the end-of-range.  That isn't calendar-sensitive.

In fact, I think it shouldn't be up to the calendar to enforce the range.  The range is an internal detail of the data model; a calendar could theoretically support a longer range.

Reply:
I will look at this tomorrow and see if anything comes to mind.

Reply:
I didn't get a chance to look at this today, but I still intend to look at it tomorrow.

Reply:
~~tomorrow~~Monday

Reply:
I have found something for this, but it's fairly horrifying and involves an extra observable call to dateFromFields: https://github.com/tc39/proposal-temporal/pull/638/files#diff-59e7894a585080160421e02806e49b00R590-R606

> In fact, I think it shouldn't be up to the calendar to enforce the range.

I don't believe that's possible, at least not in the form that we have things now. The calendar has to call the constructor, and the constructor has to reject out-of-range values. If we changed the `dateFromFields` etc. methods to return a property bag of ISO fields, instead of a Temporal object, then that might be possible to achieve.

Reply:
The calendar should just let errors from the constructor bubble up, right?  If someone wanted to support dates further in the past or future, they could subclass Temporal.DateTime, pass it to the calendar, and the calendar should do its best to succeed.

I guess in general I really don't understand the "clipping" behavior at the edge.  If a date is out of range, I would either expect it to throw, or be set to some constant infinity value, even if disambiguation is constrain.

In other words, with disambiguation: "constrain", I think the calendar should be responsible for constraining to make sure the fields are self-consistent (not too many days in the month, etc), but not constraining based on arbitrary limits set on the particular Temporal.DateTime implementation.

Can we discuss this point in the champions call this week?

For now, I think check in what you have (no dateTimeFromFields method), and we can decide later whether to add that method back again, or implement some other solution to this problem.  Does that sound reasonable?

Reply:
Sounds good. I put a link to this comment on the agenda.

Reply:
→ #664